src/App.tsx

```diff
@@ -22,54 +23,161 @@ enum SortType {
 }
 
 // Use this function in the render method
-function getReorderedGoods(
-  goods: string[],
-  sortType: SortType,
-  isReversed: boolean,
-) {
-  // Not to mutate the original array
-  const visibleGoods = [...goods];
-
-  // Sort and reverse goods if needed
-  // ...
-
-  return visibleGoods;
-}
 
 // DON'T save goods to the state
 type State = {
   isStarted: boolean,
   isReversed: boolean,
   sortType: SortType,
+  activeSortByAlphabet: boolean | null,
+  activeSortByLength: boolean | null
 };
 
-export const App = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export class App extends React.Component <{}, State> {
+  state = {
+    isStarted: false,
+    isReversed: false,
+    sortType: 0,
+    activeSortByAlphabet: false,
+    activeSortByLength: false,
+  };
+
+  start = () => {
+    this.setState({ isStarted: true });
+    this.forceUpdate();
+  };
+
+  sortByAlphabet = () => {
+    this.setState(state => ({
+      sortType: SortType.ALPABET,
+      activeSortByAlphabet: !state.activeSortByAlphabet,
+    }));
+  };
+
+  sortByLength = () => {
+    this.setState(state => ({
+      sortType: SortType.LENGTH,
+      activeSortByLength: !state.activeSortByLength,
+    }));
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortType: 0,
+    });
+  };
+
+  render() {
+    const {
+      isStarted,
+      isReversed,
+      sortType,
+      activeSortByAlphabet,
+      activeSortByLength,
+    } = this.state;
+    const goods = [...goodsFromServer];
+
+    goods.sort((item1, item2) => {
+      switch (sortType) {
+        case SortType.ALPABET:
+          return item1.localeCompare(item2);
+        case SortType.LENGTH:
+          return (item1.length - item2.length);
+        default:
+          return 0;
+      }
+    });
```

Comment:
Your sort will start every time and will check each element and do nothing
```suggestion
    switch (sortType) {
      case SortType.ALPABET:
        return visibleGoods.sort((a, b) => a.localeCompare(b));
      case SortType.LENGTH:
        return visibleGoods.sort((a, b) => a.length - b.length);
      case SortType.NONE:
      default:
        break;
    }
```